src/components/table/OcTableFiles.vue

```diff
@@ -388,6 +390,46 @@ export default {
 
       return this.$gettext("Select file")
     },
+
+    getSharedWithAvatarDescription(resource) {
+      const resourceType =
+        resource.type === "folder" ? this.$gettext("folder") : this.$gettext("file")
+
+      const shareCount = resource.sharedWith.filter(u => !u.link).length
+      const linkCount = resource.sharedWith.filter(u => !!u.link).length
+
+      const baseText = this.$gettext("This %{ resourceType } is shared via ")
+      const shareText =
```

Comment:
Argh. This doesn't work. Gettext extracts the strings. Concatenating the **results** (!) of gettext will provide correct results for English, but not for other languages that have different grammar...

Reply:
Meaning: All combinations / concatenations need to be full strings.

Reply:
Updated the code, is this the pragmatic solution you were looking after? 🤔 

And does `$gettextInterpolate` complain about unused variables?

Reply:
"Explanation":

```js
switch (true) {
  case shareCount == 1 && linkCount == 0:
    // one share no link
    description = this.$gettext("This %{ resourceType } is shared via one invite")
    break
  case shareCount == 0 && linkCount == 1:
    // no share one link
    description = this.$gettext("This %{ resourceType } is shared via one link")
    break
  case shareCount == 1 && linkCount == 1:
    // one share one link
    description = this.$gettext(
      "This %{ resourceType } is shared via one invite and one link"
    )
    break
  case shareCount == 0 && linkCount > 1:
    // no share many links
    description = this.$gettext("This %{ resourceType } is shared via %{ linkCount } link")
    break
  case shareCount == 1 && linkCount > 1:
    // one share many links
    description = this.$gettext(
      "This %{ resourceType } is shared via one invite and %{ linkCount } links"
    )
    break
  case shareCount > 1 && linkCount == 0:
    // many shares no links
    description = this.$gettext(
      "This %{ resourceType } is shared via %{ shareCount } invites"
    )
    break
  case shareCount > 1 && linkCount == 1:
    // many shares one link
    description = this.$gettext(
      "This %{ resourceType } is shared via %{ shareCount } invites and one link"
    )
    break
  case shareCount > 1 && linkCount > 1:
    // many shares many links
    description = this.$gettext(
      "This %{ resourceType } is shared via %{ shareCount } invites and %{ linkCount } links"
    )
    break
}
```

Reply:
Still not happy with it. Pluralization works differently in different languages. Therefore we _must_  use `$ngettext`. But it's simply not possible to have multiple `n`s. A solution would be to have two independent sentences. One for links, one for invites. Doing proper plualization and variable injection in the full strings. And then just concatenating the fully translated independent sentences as one resulting string. I.e.

`$ngettext("This %{ resourceType } is shared via %{ linkCount } link", "This %{ resourceType } is shared via %{ linkCount } links", links.length)`.

Further reading: http://docs.translatehouse.org/projects/localization-guide/en/latest/l10n/pluralforms.html?id=l10n/pluralforms

TL;DR: don't enforce pluralization from your own language. :trollface: Sorry for being a PITA with this.

Reply:
Okay so I kinda reverted/refactored the changes before the `switch-case`-piece of art and rebased to master, let me know what you think about this one!